spec/integrations/shared/base_render_examples.rb

```diff
@@ -50,6 +50,74 @@ def extract(field_name, object, _local_options, _options={})
     it('returns json derived from a custom extractor') { should eq(result) }
   end
 
+  context 'Given blueprint has ::field with a :if argument' do
+    variants = %i[proc method].product([true, false])
+
+    variants.each do |type, value|
+      context "#{type} returning #{value}" do
+        let(:if_value) { value }
+        let(:unless_value) { false }
+        let(:field_options) { {} }
+        let(:local_options) { { x: 1, y: 2 } }
+        let(:if_proc) { ->(_obj, _local_opts) { if_value } }
+        let(:unless_proc) { ->(_obj, _local_opts) { unless_value } }
+        let(:blueprint) do
+          f_options = field_options
+
+          bp = Class.new(Blueprinter::Base) do
+            field :id
+            field :first_name, f_options
+          end
+          bp.instance_eval <<-RUBY, __FILE__, __LINE__ + 1
+            def self.if_method(_object, _options)
+              #{if_value}
+            end
+
+            def self.unless_method(_object, _options)
+              #{unless_value}
+            end
+          RUBY
+          bp
+        end
+
+        let(:result_with_first_name) do
+          %({"first_name":"Meg","id":#{obj_id}})
+        end
+        let(:result_without_first_name) { %({"id":#{obj_id}}) }
+        subject { blueprint.render(obj, local_options) }
+
+        before do
+          field_options[:if] = type == :method ? :if_method : if_proc
+        end
+
+        context 'and no :unless argument' do
+          if value
+            it { should eq(result_with_first_name) }
+          else
+            it { should eq(result_without_first_name) }
+          end
+        end
+
+        variants.each do |other_type, other_value|
+          context "and :unless is #{other_type} returning #{other_value}" do
+            let(:unless_value) { other_value }
+            before do
+              field_options[:unless] = if type == :method then :unless_method
+                                       else unless_proc
+                                       end
+            end
+
+            if value && !other_value
+              it { should eq(result_with_first_name) }
```

Comment:
The current rspec printout out for these `it` blocks are something like:
```
and :unless is proc returning false
  should eq "{\"first_name\":\"Meg\",\"id\":15}"
```
or
```
and :unless is proc returning true
  should eq "{\"id\":16}"`
```

When reading the rspec printout, we don't actually care whether `first_name` is there or not, as that is just the example. What we care is whether the field with the conditional is there or not.

I think a description for the `it` block should be there to clarify the printout. 
`it ('does not serialize the conditional field') { should eq(result_without_first_name) }`
or
`it ('serializes the conditional field') { should eq(result_with_first_name) }`

which gives a printout of something like:

```
and :unless is proc returning true
  serialize the conditional field
```
or
```
and :unless is proc returning false
  does not serialize the conditional field
```

This comment applies to the other `it` blocks in this PR as well.